src/main/java/de/hpi/bpt/argos/parsing/EventTypeParserImpl.java

```diff
@@ -0,0 +1,221 @@
+import java.util.Map;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public class EventTypeParserImpl implements EventTypeParser {
+	private static final Logger logger = LoggerFactory.getLogger(EventTypeParserImpl.class);
+	private static final JsonParser jsonParser = new JsonParser();
+
+	private static final String JSON_NAME = "name";
+	private static final String JSON_TIMESTAMP = "timestamp";
+	private static final String JSON_ATTRIBUTES = "attributes";
+
+	private static EventTypeParser instance;
+
+	/**
+	 * This constructor hides the default public one to implement the singleton pattern.
+	 */
+	private EventTypeParserImpl() {
+
+		/**
+		 * File format from unicorn
+		 * Please note, that the given data-types are ignored.
+		 *
+		 * {
+		 *     "name":"EventTypeName",
+		 *     "timestamp":"EventTypeTimestamp",
+		 *     "attributes":{
+		 *			"eventTypeAttribute1":"STRING",
+		 *			"eventTypeAttribute2":"STRING",
+		 *			"eventTypeAttribute3":"INTEGER",
+		 *     }
+		 * }
+		 */
+	}
+
+	/**
+	 * This method returns the singleton instance of this class.
+	 * @return - the singleton instance of this class
+	 */
+	public static EventTypeParser getInstance() {
+		if (instance == null) {
+			instance = new EventTypeParserImpl();
+		}
+
+		return instance;
+	}
+
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void loadEventTypes() {
+		String eventTypesDirectoryPath = Argos.getEventTypesDirectory();
+		if (eventTypesDirectoryPath.length() == 0) {
+			logger.info("event types directory not defined in properties");
+			return;
+		}
+
+		// since the path is delivered as URI, it is represented as a HTML string. Thus we need to replace %20 with file system spaces.
+		File eventTypesDirectory = new File(eventTypesDirectoryPath.replaceAll("%20", " "));
+
+		try {
+			for (File eventTypeFile : eventTypesDirectory.listFiles()) {
+				if (!eventTypeFile.getName().endsWith(".json")) {
+					continue;
+				}
+
+				loadDefaultEventType(eventTypeFile);
+			}
+
+		} catch (NullPointerException e) {
+			LoggerUtilImpl.getInstance().error(logger, "cannot find event types directory", e);
+		}
+	}
+
+	/**
+	 * This method loads the content of a specific file and parses this into an event type.
+	 * @param eventTypeFile - the file to load
+	 */
+	private void loadDefaultEventType(File eventTypeFile) {
+		try {
+			if (!wasModified(eventTypeFile)) {
+				logger.info(String.format("file '%1$s' was loaded already and therefore skipped", eventTypeFile.getName()));
+				return;
+			} else {
+				modify(eventTypeFile);
+			}
+
+			String fileContent = new String(Files.readAllBytes(Paths.get(eventTypeFile.toURI())), StandardCharsets.UTF_8);
+
+			JsonObject jsonEventType = jsonParser.parse(fileContent).getAsJsonObject();
+
+			if (jsonEventType.get(JSON_NAME).getAsString().isEmpty()) {
+				return;
+			}
+
+			List<EventType> eventTypes = PersistenceAdapterImpl.getInstance().getEventTypes();
+
+			for (EventType eventType : eventTypes) {
```

Comment:
A comment why you check this would probably be helpful. (Because event type names should not be doubled?)